llvm/lib/IR/Constants.cpp

```diff
@@ -885,6 +901,26 @@ ConstantInt *ConstantInt::get(LLVMContext &Context, const APInt &V) {
   return Slot.get();
 }
 
+// Get a ConstantInt vector with each lane set to the same APInt.
+ConstantInt *ConstantInt::get(LLVMContext &Context, ElementCount EC,
+                              const APInt &V) {
```

Comment:
If you implement them this way, you need to make the APIs private. Having this as a public API means that it should return old splat representation if the relevant options are disabled.

Reply:
I've made them private but had to add VectorConstant to the friends list so that it can use them.